packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -131,6 +106,37 @@ bool ArbCore::checkError() {
     return core_error;
 }
 
+void ArbCore::printMachineOutputInfo(const std::string& msg,
+                                     MachineOutput& machine_output) {
+    std::string l2_block_string;
+    if (machine_output.l2_block_number > coreConfig.final_block) {
+        l2_block_string = "TERMINAL_BLOCK";
```

Comment:
Won't this always print TERMINAL_BLOCK if final_block is the default of zero?

Reply:
Fixed